objects/src/accounts/account_id.rs

```diff
@@ -40,39 +63,242 @@ pub enum AccountType {
+            REGULAR_ACCOUNT_IMMUTABLE_CODE => Ok(AccountType::RegularAccountImmutableCode),
+            FUNGIBLE_FAUCET => Ok(AccountType::FungibleFaucet),
+            NON_FUNGIBLE_FAUCET => Ok(AccountType::NonFungibleFaucet),
+            v => Err(AccountError::InvalidAccountType(v)),
+        }
+    }
+}
 
-pub const ON_CHAIN: u64 = 0b00;
-pub const OFF_CHAIN: u64 = 0b10;
+impl TryFrom<Felt> for AccountType {
+    type Error = AccountError;
+
+    fn try_from(value: Felt) -> Result<Self, AccountError> {
+        value.as_int().try_into()
+    }
+}
+
+impl From<AccountType> for u64 {
+    fn from(value: AccountType) -> Self {
+        value as u64
+    }
+}
+
+// ACCOUNT STORAGE TYPE
+// ================================================================================================
 
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
 #[repr(u64)]
 pub enum AccountStorageType {
-    OnChain = ON_CHAIN,
-    OffChain = OFF_CHAIN,
+    OnChain = ACCOUNT_ON_CHAIN,
+    OffChain = ACCOUNT_OFF_CHAIN,
+}
+
+impl Display for AccountStorageType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            AccountStorageType::OnChain => write!(f, "on-chain"),
+            AccountStorageType::OffChain => write!(f, "off-chain"),
+        }
+    }
+}
+
+impl TryFrom<u64> for AccountStorageType {
+    type Error = AccountError;
+
+    fn try_from(value: u64) -> Result<Self, Self::Error> {
+        match value {
+            ACCOUNT_ON_CHAIN => Ok(AccountStorageType::OnChain),
+            ACCOUNT_OFF_CHAIN => Ok(AccountStorageType::OffChain),
+            v => Err(AccountError::InvalidStorageType(v)),
+        }
+    }
+}
+
+impl TryFrom<Felt> for AccountStorageType {
+    type Error = AccountError;
+
+    fn try_from(value: Felt) -> Result<Self, Self::Error> {
+        value.as_int().try_into()
+    }
+}
+
+impl From<AccountStorageType> for u64 {
+    fn from(value: AccountStorageType) -> Self {
+        value as u64
+    }
+}
+
+// ACCOUNT POW
+// ================================================================================================
+
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct AccountPoW(u8);
+
+impl AccountPoW {
+    pub fn new(pow: u8) -> Result<Self, AccountError> {
+        if u64::from(pow) > ACCOUNT_POW_MAXIMUM {
+            Err(AccountError::InvalidPoW(u64::from(pow)))
+        } else {
+            Ok(AccountPoW(pow))
+        }
+    }
+
+    /// Returns the value configured for the proof-of-work.
+    pub fn as_int(&self) -> u8 {
+        self.0
+    }
+}
+
+impl Display for AccountPoW {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "AccountPow({})", self.0)
+    }
+}
+
+impl TryFrom<u64> for AccountPoW {
```

Comment:
Note: These types are using `u64` conversion to and from their **mask position**! This is confusing at times, I had a few instances that tried to PoW 80bits because I used `u64::from(pow)` instead of `pow.as_int()`